include/aspect/melt.h

```diff
@@ -0,0 +1,329 @@
+                            const typename MaterialModel::Interface<dim>::MaterialModelOutputs &material_model_outputs,
+                            const typename Simulator<dim>::AdvectionField &advection_field,
+                            const unsigned int q_point) const;
+    };
+
+    /**
+     * A namespace for the definition of functions that implement various
+     * other terms that need to occasionally or always be assembled.
+     */
+    namespace OtherTerms
+    {
+      /**
+       * Integrate the local fluid pressure shape functions on a single cell
+       * for models with melt migration, so that they can later be used to do
+       * the pressure right-hand side compatibility modification.
+       */
+      template <int dim>
+      void
+      pressure_rhs_compatibility_modification_melt (const SimulatorAccess<dim>                      &simulator_access,
+                                                    internal::Assembly::Scratch::StokesSystem<dim>  &scratch,
+                                                    internal::Assembly::CopyData::StokesSystem<dim> &data);
+    }
+  }
+
+
+  /**
+   * Class to contain all runtime parameters and other helper functions
+   * related to melt transport. A global instance can be retrieved with
+   * MeltHandler<dim>::get(), but keep in mind that it only exists if
+   * parameters.include_melt_transport is true.
+   */
+  template <int dim>
+  class MeltHandler: public SimulatorAccess<dim>
+  {
+    public:
+      MeltHandler(ParameterHandler &prm);
+
+      /**
+       * Declare additional parameters that are needed in models with
+       * melt transport (including the fluid pressure boundary conditions).
+       */
+      static void declare_parameters (ParameterHandler &prm);
+
+      /**
+       * Parse additional parameters that are needed in models with
+       * melt transport (including the fluid pressure boundary conditions).
+       *
+       * This has to be called before edit_finite_element_variables,
+       * so that the finite elements that are used for the additional melt
+       * variables can be specified in the input file and are parsed before
+       * the introspection object is created.
+       */
+      void parse_parameters (ParameterHandler &prm);
+
+      /**
+       * Add the additional variables we need in simulations with melt
+       * migration to the list of variables, which will be used later
+       * to set up the introspection object.
+       */
+      void edit_finite_element_variables(const Parameters<dim> &parameters, std::vector<VariableDeclaration<dim> > &variables);
+
+      /**
+       * Setup SimulatorAccess for the plugins related to melt transport.
+       */
+      void initialize_simulator (const Simulator<dim> &simulator_object);
+
+      /**
+       * Compute fluid velocity and solid pressure in this ghosted solution vector.
+       * @param solution
+       */
+      void compute_melt_variables(LinearAlgebra::BlockVector &solution);
```

Comment:
I suspect you wanted to document `@param solution`? Is this the existing solution vector, or an auxiliary vector that will be filled with whatever you compute?

I don't know the formulation well enough, but suspect that the fluid velocity and solid pressure can be computed algebraically (i.e., without solving an equation) from the existing solution? If so, maybe say so here.